db/collection.go

```diff
@@ -401,7 +404,12 @@ func getSchemaTypeAtPath(schema *jsonschema.Schema, pth string) (*jsonschema.Typ
 	return jt, nil
 }
 
+// getSchemaTypeProperties extracts a map of schema properties from a given input schema.
+// If there are no available properties, it will return an empty (nil) map
```

Comment:
What's your opinion about returning an empty map (`make(map[string]*jsonschema.Type`)?
Returning `error = nil`, and a `nil` map may be buying a future panic on someone trying to use the map.

Reply:
Ah yeh, that's not a bad idea, so calls like map["something"] will always return empty, but at least not panic. Good idea. Will push that change and then merge.

Reply:
Just in case, in Go you can do:
```go
var foo map[string]string // So, nil map
fmt.Println(foo["hello"))
```
and that won't panic.

The problem with nil in maps is writing (e.g: `foo["bad"]="idea"`), you'll get a panic.
So the comment was more about just in case covering that scenario, which most prob the caller shouldn't do (but just in case).